src/orquestra/quantum/wip/operators/_pauli_operators.py

```diff
@@ -83,58 +84,94 @@ def _validate_type(object: Any) -> None:
         )
 
 
+def _is_in_brackets(string: str) -> bool:
+    return string.startswith("(") and string.endswith(")")
+
+
+def _parse_complex(complex_str: str) -> complex:
+    value = complex(complex_str.replace(" ", ""))
+    if value.real != 0 and value.imag != 0 and not _is_in_brackets(complex_str):
+        raise ValueError(
+            "Complex number with nonzero real and imaginary part has to be "
+            "enclosed in bracket."
+        )
+    return value
+
+
+def _parse_operator(op_str: str) -> Tuple[int, str]:
+    match = re.match(r"([XYZI])([0-9]+)$", op_str, re.I)
+
+    if not match:
+        raise ValueError("Badly formatted string representation passed.")
+
+    return int(match.group(2)), match.group(1).upper()
+
+
+def _parse_operators_and_coefficient(
+    term_str: str,
+) -> Tuple[Optional[complex], Dict[int, str]]:
+    parts = re.split(r"\ *\*\ *", term_str.strip(" "))
+    try:
+        coef = _parse_complex(parts[0])
+        operators_strs = parts[1:]
+    except ValueError:
+        coef = None
+        operators_strs = parts
+
+    operators_dict = dict([_parse_operator(op_str) for op_str in operators_strs])
+
+    if len(operators_dict) != len(operators_strs):
+        raise ValueError("Duplicate qubit index in a term detected.")
+
+    return coef, dict([_parse_operator(op_str) for op_str in operators_strs])
+
+
 class PauliTerm:
-    """
-    A datastructure for storing information about a single Pauli Term
-    """
+    """Representation of a single Pauli Term."""
 
     def __init__(
         self,
         operator: Union[str, Dict[int, str]],
-        coefficient: complex = 1.0,
+        coefficient: Optional[complex] = None,
```

Comment:
[nitpick] maybe add a note that the default coefficient is 1 now that it's no longer clear from reading the default arguments?

Reply:
Added in [c8c6056](https://github.com/zapatacomputing/orquestra-quantum/pull/44/commits/c8c605691a35ff4232d0d6c42c5690e3c9e859c1)